### DISCUSSION TITLE
highscores/scoring

### DISCUSSION THREAD

**Discussion by jbrokamp** [plbrault/youre-the-os#212] [Ideas]
Some questions about scores:

1 .Some way to keep track of (personal) high scores?
Either stored client-side (web storage), or on itch.io / server?

2. Another thing is, that I get way higher score on EASY (500k, 10 minutes) than on NORMAL (150k, 4 minutes), it could be nice to give higher rewards on harder difficulties, to incentivize the harder difficulties?

3. Bonus points when you gracefully terminate a process? Gives some incentive to focus more on processes with lower id, also clears out RAM more efficiently that way. Maybe even visualize the progress of a process somehow.

**Comment by Wiguwbe** [comment]
Hey,

I don't know about 1),

Regarding 2), not knowing about the scoring method, either way I think there should always be a differentiation between the difficulties.
Even if the scoring was _leveled_, it probably is hard to compare between difficulty levels.

For 3), it looks like that, if the process is green and in CPU, for each tick, there is a 1% change of the process terminating.
I do agree that a process having a predetermined number of ticks to run could add another level to the game :)

**Comment by plbrault** [comment]
Not sure about visualising the progress of a process towards graceful termination, as an OS does not know when a process is going to end. Also, right now the player does get 1000 extra points whenever a process gracefully terminates.

**Comment by plbrault** [comment]
There seems to be a way with pygbag to use the browser's local storage: https://pygame-web.github.io/wiki/pygbag-code/#handling-persistent-data-across-sessions

This would probably be the way to go to implement local high scores in the browser.
